tests/client/test_client.py

```diff
@@ -38,7 +38,7 @@ def test_can_construct_with_throttle_arg(requests_mocker):
 
 def test_can_construct_with_session_args(requests_mocker):
     """A client instance can be constructed with requests.Session kwargs."""
-    client = Client("https://pulp.example.com/", auth=("x", "y"), verify=False)
+    client = Client("https://pulp.example.com/", headers={"x": "y"}, verify=False)
```

Comment:
Given that you're implementing special behavior for `headers` argument, a proper test which covers that behavior would be nice.  This change seems the bare minimum to satisfy the coverage tool, only verifying that it doesn't crash.